pkgs/development/libraries/libcanberra/default.nix

```diff
@@ -23,7 +23,9 @@ stdenv.mkDerivation rec {
   configureFlags = [ "--disable-oss" ];
 
   patchFlags = "-p0";
-  patches = stdenv.lib.optional stdenv.isDarwin
+  patches = [
+    ./gtk_dont_assume_x11.patch
```

Comment:
Did you had to modify this patch in order to apply it?
If yes, please include the link to the commit in a comment above: http://git.0pointer.net/libcanberra.git/patch/?id=c0620e432650e81062c1967cc669829dbd29b310
Otherwise please use `fetchpatch { url = "http://git.0pointer.net/libcanberra.git/patch/?id=c0620e432650e81062c1967cc669829dbd29b310"; sha256 = "..."; }`

Reply:
It seems like stdenv doesn't like git-style patches with a/ and b/. This is why I modified the patch to unix {.orig,} style. Is there a way to use git-style patches with stdenv?

Reply:
Pulling patches from upstream makes it obvious that we can drop them in the next version and we don't need to include the content of them in our repository.

Reply:
There is `patchFlags`. i.e. enabling/disabling stripping of prefixes can be done with `patchFlags = [ "-p1" ]`.

Reply:
Since there is another patch in unix-format, I would just add the linke to git.0pointer.net to the commit description.

Reply:
The other patch requires -p0. I doubt it's possible to set this option per-patch, so I will leave the content as it is and add the link to the commit description.